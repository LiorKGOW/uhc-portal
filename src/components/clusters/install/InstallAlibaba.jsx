import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  PageSection,
} from '@patternfly/react-core';

import Breadcrumbs from '../../common/Breadcrumbs';
import { tollboothActions } from '../../../redux/actions';
import { scrollToTop } from '../../../common/helpers';
import instructionsMapping from './instructions/instructionsMapping';
import OCPInstructions from './instructions/OCPInstructions';
import PageTitle from '../../common/PageTitle';

export class InstallAlibaba extends Component {
  componentDidMount() {
    scrollToTop();
    document.title = 'Install OpenShift 4 | Red Hat OpenShift Cluster Manager | Alibaba';

    const { dispatch } = this.props;
    dispatch(tollboothActions.createAuthToken());
  }

  render() {
    const { token } = this.props;
    const breadcrumbs = (
      <Breadcrumbs path={[
        { label: 'Clusters' },
        { label: 'Create', path: '/create' },
        { label: 'Alibaba' },
      ]}
      />
    );

    return (
      <>
        <PageTitle title={instructionsMapping.alibaba.title} breadcrumbs={breadcrumbs} />
        <PageSection>
          <OCPInstructions
            token={token}
            cloudProviderID="alibaba"
            {...instructionsMapping.alibaba}
          />
        </PageSection>
      </>
    );
  }
}

InstallAlibaba.propTypes = {
  token: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({ token: state.tollbooth.token });

export default connect(mapStateToProps)(InstallAlibaba);
