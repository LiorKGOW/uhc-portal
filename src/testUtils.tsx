import React from 'react';
import { Provider } from 'react-redux';
import { createStore, Store } from 'redux';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import { toHaveNoViolations, axe } from 'jest-axe';

import { createBrowserHistory } from 'history';
import { store } from './redux/store';
import { reduxReducers } from './redux/reducers';
import * as restrictedEnv from './restrictedEnv';

const history = createBrowserHistory();

export const TestWrapper = ({
  children,
  initialStore,
}: {
  children: React.ReactNode;
  initialStore?: Store;
}) => <Provider store={initialStore || store}>{children}</Provider>;

const renderWithState = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  initialState?: any, // setting to any because only a partial state structure may be sent
) => {
  const newStore = initialState ? createStore(reduxReducers(history), initialState) : store;

  const Wrapper = ({ children }: { children: React.ReactNode }) => (
    <TestWrapper initialStore={newStore}>{children}</TestWrapper>
  );
  return { ...render(ui, { wrapper: Wrapper, ...options }), user: userEvent.setup() };
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// Type not exported in the library
export type UserEventType = {
  type: (
    field: HTMLElement,
    typeValue: string,
    { initialSelectionStart }: { initialSelectionStart: number },
  ) => void;
  clear: (field: HTMLElement) => void;
};

export { renderWithState as render };

/* ***** Items outside of React Test Library ************ */
expect.extend(toHaveNoViolations);

export const checkAccessibility = async (container: HTMLElement | string) => {
  const results = await axe(container);
  expect(results).toHaveNoViolations();
};

export const insightsMock = () => {
  global.insights = {
    chrome: {
      ...global.insights.chrome,
      on: () => () => {},
      appNavClick: () => {},
      auth: {
        getUser: () => Promise.resolve({ data: {} }),
        getToken: () => Promise.resolve(),
        getOfflineToken: () => Promise.resolve({ data: { refresh_token: 'hello' } }),
      },
    },
  };
};

export const mockRestrictedEnv = () => {
  const mock = jest.spyOn(restrictedEnv, 'isRestrictedEnv');
  mock.mockReturnValue(false);
  return mock;
};
