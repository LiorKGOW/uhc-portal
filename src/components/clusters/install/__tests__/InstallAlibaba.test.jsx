import React from 'react';

import { checkAccessibility, screen, withState } from '~/testUtils';

import githubReleases from '../githubReleases.mock';
import { InstallAlibaba } from '../InstallAlibaba';

jest.mock('../../../../redux/actions', () => ({
  __esModule: true,
  tollboothActions: {
    createAuthToken: jest.fn().mockResolvedValue('foo'),
  },
  githubActions: {
    getLatestRelease: jest.fn(),
  },
}));

describe('InstallAlibaba', () => {
  const dispatch = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });
  it('is accessible', async () => {
    const { container } = withState(githubReleases).render(
      <InstallAlibaba token={{}} dispatch={dispatch} />,
    );

    expect(
      await screen.findByText(
        'Follow the documentation to configure your Alibaba Cloud account and run the installer',
      ),
    ).toBeInTheDocument();
    await checkAccessibility(container);
  });
});
