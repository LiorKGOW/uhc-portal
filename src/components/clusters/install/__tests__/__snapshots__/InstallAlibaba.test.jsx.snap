// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallAlibaba renders correctly 1`] = `
<Fragment>
  <PageTitle
    breadcrumbs={
      <Breadcrumbs
        path={
          Array [
            Object {
              "label": "Clusters",
            },
            Object {
              "label": "Cluster Type",
              "path": "/create",
            },
            Object {
              "label": "Alibaba Cloud",
            },
          ]
        }
      />
    }
    title="Install OpenShift on Alibaba Cloud with installer-provisioned infrastructure"
  />
  <PageSection>
    <OCPInstructions
      channel="stable"
      cloudProvider="Alibaba Cloud"
      cloudProviderID="alibaba"
      customizations="https://docs.openshift.com/container-platform/4.13/installing/installing_alibaba/installing-alibaba-customizations.html"
      docURL="https://docs.openshift.com/container-platform/4.13/installing/installing_alibaba/preparing-to-install-on-alibaba.html"
      installer="x86_64-openshift-install"
      isBMIPI={false}
      isUPI={false}
      publicCloud={true}
      title="Install OpenShift on Alibaba Cloud with installer-provisioned infrastructure"
      token={Object {}}
    />
  </PageSection>
</Fragment>
`;
